### PR TITLE
Change prompt to ginga2

### PR DESCRIPTION
**Pull request by nilnil47**

### CODE REVIEW

File: auto_post_classifier/main.py [hunk 1]
```diff
@@ -1,62 +1,64 @@
-import json
+import logging
```
Comment (drorata) [inline auto_post_classifier/main.py:1]: Do you need both `logging` and `loguru`?

File: auto_post_classifier/main.py [hunk 2]
```diff
@@ -1,62 +1,64 @@
-import json
```
Comment (drorata) [inline auto_post_classifier/main.py:1]: Once again, only linting, right?

File: auto_post_classifier/response_persister.py [hunk 3]
```diff
@@ -1,27 +1,24 @@
 import csv
```
Comment (drorata) [inline auto_post_classifier/response_persister.py:1]: Linting changes only, correct?

**Review by drorata — COMMENTED** [review]
I left a few comment. I didn't yet review the following files:

- `auto_post_classifier/test_main.py`
- `auto_post_classifier/prompts/gpt3_5_user.prompt`
- `auto_post_classifier/gpt_handler.py`
- `auto_post_classifier/test_main.py`

I feel it'll be fast if we either split these actual changes into a separate PR or do it live quickly.

File: auto_post_classifier/gpt_handler.py [hunk 4]
```diff
@@ -42,140 +41,116 @@ def validate_json(self, response_dict: dict):
...
+        self, user_template_path: str, system_template_path: str
+    ):
+        self.user_prompt_template = self.jinja_environment.get_template(
+            user_template_path
+        )
+        self.system_prompt_template = self.jinja_environment.get_template(
+            system_template_path
+        )
+
+    def __init__(
```
Comment (drorata) [inline auto_post_classifier/gpt_handler.py:81]: Move to the beginning of the class.

**Review by drorata — APPROVED** [review]

File: Makefile [hunk 5]
```diff
@@ -1,5 +1,9 @@
-.PHONY: code_style_validation
+.PHONY: code_style_validation code_style_fixing
```
[the remark this answers is not in the document]
Comment (nilnil47) [reply Makefile:1]: fixed

File: auto_post_classifier/api.py [hunk 6]
```diff
@@ -1,70 +1,73 @@
-import json
```
[the remark this answers is not in the document]
Comment (nilnil47) [reply auto_post_classifier/api.py:1]: טקד

File: auto_post_classifier/testing_utils.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import json
```
[the remark this answers is not in the document]
Comment (nilnil47) [reply auto_post_classifier/testing_utils.py:1]: added documentation

File: auto_post_classifier/main.py (hunk 1 above)
Comment (nilnil47) [reply auto_post_classifier/main.py:1]: fixed, remove also the 422 response code handle

File: auto_post_classifier/response_persister.py (hunk 3 above)
Comment (nilnil47) [reply auto_post_classifier/response_persister.py:1]: yes

**Pull request merged**